### PR TITLE
Snowflake source freshness

### LINKED ISSUE #4876: Add caveat for `dbt-snowflake` that the source data freshness may reflect non-data changes
### Contributions

- [ ] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

Put details on: https://docs.getdbt.com/reference/resource-configs/snowflake-configs

Link to these details from: https://docs.getdbt.com/docs/deploy/source-freshness

### What part(s) of the page would you like to see updated?

Add a new section for Source Freshness on: https://docs.getdbt.com/reference/resource-configs/snowflake-configs

Add a new section for platform-specific limitations on: https://docs.getdbt.com/docs/deploy/source-freshness.

### Additional information

This gap in docs was identified in https://github.com/dbt-labs/dbt-snowflake/issues/899.

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Closes https://github.com/dbt-labs/docs.getdbt.com/issues/4876

### CODE REVIEW

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 1]
```diff
@@ -479,3 +479,17 @@ The workaround is to execute `DROP TABLE my_model` on the data warehouse before
...
+Snowflake source freshness is calculated using information from the `LAST_ALTERED` column. 
+
+Per the [Snowflake documentation](https://docs.snowflake.com/en/sql-reference/info-schema/tables#usage-notes): 
+
+  >The `LAST_ALTERED` column is updated when the following operations are performed on an object:
+  >- DDL operations.
+  >- DML operations (for tables only).
+  >- Background maintenance operations on metadata performed by Snowflake.
+
+This means that Snowflake source frehssness is determined by a field that is updated when any object is updated, not just data updates. There is no action that needs to be taken, but analytics teams should note this caveat.
```
Comment (matthewshaver) [inline website/docs/reference/resource-configs/snowflake-configs.md]:
```suggestion
This means that Snowflake source freshness is determined by a field that is updated when any object is updated, not just data updates. No action must be taken, but analytics teams should note this caveat. 
```

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 2]
```diff
@@ -479,3 +479,17 @@ The workaround is to execute `DROP TABLE my_model` on the data warehouse before
 </VersionBlock>
 
 </VersionBlock>
+
+
+## Source freshness known limitation
+
+Snowflake source freshness is calculated using information from the `LAST_ALTERED` column.
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/snowflake-configs.md]: I suggest we combine the last paragraph and the 1st. Maybe something like:
```suggestion
Snowflake calculates source freshness using information from the `LAST_ALTERED` column, meaning it relies on a field updated whenever any object undergoes modification, not only data updates. No action must be taken, but analytics teams should note this caveat. 
```

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 3]
```diff
@@ -479,3 +479,17 @@ The workaround is to execute `DROP TABLE my_model` on the data warehouse before
...
+Snowflake source freshness is calculated using information from the `LAST_ALTERED` column. 
+
+Per the [Snowflake documentation](https://docs.snowflake.com/en/sql-reference/info-schema/tables#usage-notes): 
+
+  >The `LAST_ALTERED` column is updated when the following operations are performed on an object:
+  >- DDL operations.
+  >- DML operations (for tables only).
+  >- Background maintenance operations on metadata performed by Snowflake.
+
+This means that Snowflake source freshness is determined by a field that is updated when any object is updated, not just data updates. No action must be taken, but analytics teams should note this caveat.
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/snowflake-configs.md]:
```suggestion
```

**Review by runleonarun — COMMENTED** [review]
@matthewshaver suggestion to combine content and make sentences active.

File: website/docs/reference/resource-configs/snowflake-configs.md (hunk 3 above)
Comment (matthewshaver) [reply website/docs/reference/resource-configs/snowflake-configs.md]: Good call!

**Review by runleonarun — APPROVED** [review]
🚢 it!

**Pull request merged**